core/src/main/scala/io/qbeast/core/model/QbeastColumnStats.scala

```diff
@@ -0,0 +1,127 @@
+package io.qbeast.core.model
+
+import io.qbeast.core.transform.CDFNumericQuantilesTransformer
+import io.qbeast.core.transform.CDFStringQuantilesTransformer
+import io.qbeast.core.transform.LinearTransformer
+import io.qbeast.core.transform.StringHistogramTransformer
+import io.qbeast.core.transform.Transformer
+import org.apache.spark.sql.types.ArrayType
+import org.apache.spark.sql.types.DoubleType
+import org.apache.spark.sql.types.StringType
+import org.apache.spark.sql.types.StructField
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.AnalysisExceptionFactory
+import org.apache.spark.sql.Row
+import org.apache.spark.sql.SparkSession
+
+/**
+ * Container for Qbeast Column Stats
+ *
+ * @param columnStatsSchema
+ *   the column stats schema
+ * @param columnStatsRow
+ *   the column stats row
+ */
+case class QbeastColumnStats(columnStatsSchema: StructType, columnStatsRow: Row)
+
+object QbeastColumnStatsBuilder {
```

Comment:
The builder design pattern follows the syntax of:
```scala
builder
.setAttr1(someAttr1)
.setAttr2(someAttr2)
.setAttr3(someAttr3)
.result()
```
Which isn't the case here.

I'd recommend using an `apply` method with a companion object for `QbeastColumnStats,` which returns `Option[QbeastColumnStats].` The output should be `None` when the requirements are not met.